library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/GetRDDOfAllElementsHandler.java

```diff
@@ -45,13 +85,139 @@
+            final RDD<Map.Entry<Key, Value>> rdd = new RFileReaderRDD(
+                    operation.getSparkSession().sparkContext(),
+                    accumuloStore.getProperties().getInstance(),
+                    accumuloStore.getProperties().getZookeepers(),
+                    accumuloStore.getProperties().getUser(),
+                    accumuloStore.getProperties().getPassword(),
+                    accumuloStore.getTableName(),
+                    context.getUser().getDataAuths(),
+                    serialisedConf);
+            return rdd.mapPartitions(new EntryIteratorToElementIterator(serialisedConf), true, ELEMENT_CLASS_TAG);
+        } catch (final IOException e) {
+            throw new OperationException("IOException serialising configuration", e);
+        }
+    }
+
+    private void addValidationIterator(final AccumuloStore accumuloStore, final Configuration conf) {
+        if (accumuloStore.getProperties().getEnableValidatorIterator()) {
+            final IteratorSetting itrSetting = accumuloStore
+                    .getKeyPackage().getIteratorFactory().getValidatorIteratorSetting(accumuloStore);
+            if (null == itrSetting) {
+                LOGGER.info("Not adding validation iterator as no validation functions are defined in the schema");
+            } else {
+                LOGGER.info("Adding validation iterator");
+                InputConfigurator.addIterator(AccumuloInputFormat.class, conf, itrSetting);
+            }
+        }
+    }
+
+    private void addAggregationIterator(final AccumuloStore accumuloStore, final Configuration conf)
+            throws IteratorSettingException {
+        if (accumuloStore.getSchema().isAggregationEnabled()) {
+            // Add aggregator iterator to table for all scopes
+            LOGGER.info("Adding aggregator iterator");
+            final IteratorSetting itrSetting = accumuloStore
+                    .getKeyPackage().getIteratorFactory().getAggregatorIteratorSetting(accumuloStore);
+            InputConfigurator.addIterator(AccumuloInputFormat.class, conf, itrSetting);
+        } else {
+            LOGGER.info("Not adding aggregator iterator as aggregation is not enabled");
+        }
+    }
+
+    public static class EntryIteratorToElementIterator
+            extends AbstractFunction1<Iterator<Map.Entry<Key, Value>>, Iterator<Element>> implements Serializable {
+        private byte[] serialisedConf;
+
+        public EntryIteratorToElementIterator(final byte[] serialisedConf) {
+            this.serialisedConf = serialisedConf;
+        }
+
+        @Override
+        public Iterator<Element> apply(final Iterator<Map.Entry<Key, Value>> entryIterator) {
+            final EntryToElement entryToElement = new EntryToElement(serialisedConf);
+            return entryIterator.map(entryToElement);
+        }
+    }
+
+    public static class EntryToElement extends AbstractFunction1<Map.Entry<Key, Value>, Element> implements Serializable {
```

Comment:
EntryToElement implements serializable, but if it is actually ever serialised it will break as both converter and view are transient. Do you need a way to reconstruct these transient fields? Perhaps store the serialisedConf as a field and then you can reconstruct the transient fields.

Reply:
It doesn't need to be Serializable so I've removed "implements Serializable".